### PR TITLE
Edr Providers can raise Request Entity Too Large HTTP status

### LINKED ISSUE #1118: EDR provider plugins should be able to fail with Request Entity Too Large
**Is your feature request related to a problem? Please describe.**
An EDR API is allowed to return [HTTP status code 413](https://docs.ogc.org/is/19-086r5/19-086r5.html#toc31), Request Entity Too Large. For someone writing a provider plugin, it would be nice to have the ability to return this error if the backend for whatever reason cannot handle a valid request.

**Describe the solution you'd like**
A new provider exception like the ones found in the base provider module `pygeoapi.provider.base` that can be raised, and which results in an HTTP status code 413.

**Additional context**
A new exception could be handled in the code [here](https://github.com/geopython/pygeoapi/blob/5e7015c2c767046af03d003a45bf4baff2518fd9/pygeoapi/api.py#L3284-L3293).

### PR DESCRIPTION
**Pull request by petergarnaes**
# Overview

Allows an EDR provider plugin to raise an exception that results in an [HTTP status code 413 - Request Entity Too Large](https://docs.ogc.org/is/19-086r5/19-086r5.html#toc31). 

The standard states that the response should:

> ... the implementation should return a message explaining the query limits imposed by the server implementation.

This is why the provider must initialize the provider exception with the `request_limit_explanation` argument, which will be propagated to the HTTP response.

# Related Issue / Discussion

Should resolve #1118

# Contributions and Licensing

(as per https://github.com/geopython/pygeoapi/blob/master/CONTRIBUTING.md#contributions-and-licensing)

- [x] I'd like to contribute [feature X|bugfix Y|docs|something else] to pygeoapi. I confirm that my contributions to pygeoapi will be compatible with the pygeoapi license guidelines at the time of contribution.
- [ ] I have already previously agreed to the pygeoapi Contributions and Licensing Guidelines

### CODE REVIEW

File: pygeoapi/api.py [hunk 1]
```diff
@@ -69,6 +69,7 @@
     ProviderGenericError, ProviderConnectionError, ProviderNotFoundError,
     ProviderInvalidDataError, ProviderInvalidQueryError, ProviderNoDataError,
     ProviderQueryError, ProviderItemNotFoundError, ProviderTypeError)
+from pygeoapi.provider.base_edr import EdrProviderRequestEntityTooLarge
```
Comment (tomkralidis) [inline pygeoapi/api.py:72]: Move to `pygeoapi.provider.base.ProviderRequestEntityTooLargeError`.

File: pygeoapi/provider/base_edr.py [hunk 2]
```diff
@@ -95,3 +95,10 @@ def query(self, **kwargs):
             return getattr(self, kwargs.get('query_type'))(**kwargs)
         except AttributeError:
             raise NotImplementedError('Query not implemented!')
+
+
+class EdrProviderRequestEntityTooLarge(ProviderGenericError):
```
Comment (tomkralidis) [inline pygeoapi/provider/base_edr.py:100]: Define in `pygeoapi.provider.base.ProviderRequestEntityTooLargeError`

File: pygeoapi/provider/base_edr.py [hunk 3]
```diff
@@ -95,3 +95,10 @@ def query(self, **kwargs):
...
+    """provider request entity too large error"""
+
+    def __init__(self, request_limit_explanation: str):
+        self.request_limit_explanation = request_limit_explanation
```
Comment (tomkralidis) [inline pygeoapi/provider/base_edr.py:104]: Do we need `request_limit_explanation`?  At runtime we would be able to get the same error message via:

```python
if 1 != 2:
    raise ProviderRequestEntityTooLargeError('Requested area too large')
```

**Review by tomkralidis — CHANGES_REQUESTED** [review]

**Comment by petergarnaes** [comment]
@tomkralidis requested changes now fixed

File: pygeoapi/provider/base.py [hunk 4]
```diff
@@ -313,3 +313,8 @@ class ProviderVersionError(ProviderGenericError):
 class ProviderInvalidDataError(ProviderGenericError):
     """provider invalid data error"""
     pass
+
+
+class ProviderRequestEntityTooLarge(ProviderGenericError):
```
Comment (tomkralidis) [inline pygeoapi/provider/base.py:318]: Rename to `ProviderRequestEntityTooLargeError`

**Review by tomkralidis — CHANGES_REQUESTED** [review]

File: pygeoapi/provider/base.py (hunk 4 above)
Comment (petergarnaes) [reply pygeoapi/provider/base.py]: Ups, missed that part, sorry!

**Comment by tomkralidis** [comment]
CI failures are unrelated.

**Review by tomkralidis — APPROVED** [review]

**Pull request merged**